### ISSUE TITLE
Commit, treestate and build date information missing in `rekor-* version` since 0.9.1

### ISSUE THREAD

**Issue by LeSuisse** [sigstore/rekor#913]
**Description**

It appears some some build related information are missing in the v0.9.1. I'm guessing it is related to the bump to Go 1.18 made in #904 

Now:
```
rekor-cli: Rekor CLI

GitVersion:    v0.9.1
GitCommit:     unknown
GitTreeState:  unknown
BuildDate:     unknown
GoVersion:     go1.18.3
Compiler:      gc
Platform:      linux/amd64
```

Before:
```
rekor-cli: Rekor CLI

GitVersion:    v0.9.0
GitCommit:     66f5c0611e77d0ea15b718b958387e2d016f910a
GitTreeState:  clean
BuildDate:     2022-06-30T13:00:39Z
GoVersion:     go1.17.11
Compiler:      gc
Platform:      linux/amd64
```

**Comment by cpanato** [comment]
thanks I will take a look and fix it

**Comment by cpanato** [comment]
humm, i just built and got the information

and the CI as well https://github.com/sigstore/rekor/runs/7258165646?check_suite_focus=true

``` 
____    _____   _  __   ___    ____             ____   _       ___
 |  _ \  | ____| | |/ /  / _ \  |  _ \           / ___| | |     |_ _|
 | |_) | |  _|   | ' /  | | | | | |_) |  _____  | |     | |      | |
 |  _ <  | |___  | . \  | |_| | |  _ <  |_____| | |___  | |___   | |
 |_| \_\ |_____| |_|\_\  \___/  |_| \_\          \____| |_____| |___|
rekor-cli: Rekor CLI
GitVersion:    25dbcf5
GitCommit:     25dbcf5d2230bc02a0b1129422bcf15382fb8705
GitTreeState:  clean
BuildDate:     2022-07-08T20:41:49
GoVersion:     go1.18.3
Compiler:      gc
Platform:      linux/amd64
```

i will run a release in my fork to check this

**Comment by cpanato** [comment]
built a fork release https://github.com/cpanato/rekor/releases/tag/v99.99.03 and seems fine

```
$ ./rekor-cli-darwin-arm64 version
  ____    _____   _  __   ___    ____             ____   _       ___
 |  _ \  | ____| | |/ /  / _ \  |  _ \           / ___| | |     |_ _|
 | |_) | |  _|   | ' /  | | | | | |_) |  _____  | |     | |      | |
 |  _ <  | |___  | . \  | |_| | |  _ <  |_____| | |___  | |___   | |
 |_| \_\ |_____| |_|\_\  \___/  |_| \_\          \____| |_____| |___|
rekor-cli: Rekor CLI

GitVersion:    v99.99.03
GitCommit:     66bc41d1a2a916317b09c90a971b5c6639d1831f
GitTreeState:  clean
BuildDate:     2022-07-09T09:41:45
GoVersion:     go1.18.3
Compiler:      gc
Platform:      darwin/arm64
```


i will make some updates and for the next release i will check this before the actual release

**Comment by LeSuisse** [comment]
FYI I noticed the issue while upgrading the package in nixpkgs (https://github.com/NixOS/nixpkgs/pull/179999) so I have a reproducible environment to trigger it if it can help.

We are currently setting those information using the same flags than the ones used in the Makefile https://github.com/sigstore/rekor/blob/25dbcf5d2230bc02a0b1129422bcf15382fb8705/Makefile#L57-L60 but taking a quick look at what `sigs.k8s.io/release-utils/version` does I'm guessing those flags should not be needed anymore with Go 1.18 since it uses `debug.BuildInfo`.

**Comment by 06kellyjac** [comment]
Yep

https://github.com/kubernetes-sigs/release-utils/blob/main/version/version.go

Using buildinfo

One problem is it relies on a `.git` dir which we dont keep due to reproducibility issues.
release-utils should probably only set the value if its not already set via ldflags?

**Comment by cpanato** [comment]
fixed in v0.10.0

```
$ ./rekor-cli-darwin-arm64 version
  ____    _____   _  __   ___    ____             ____   _       ___
 |  _ \  | ____| | |/ /  / _ \  |  _ \           / ___| | |     |_ _|
 | |_) | |  _|   | ' /  | | | | | |_) |  _____  | |     | |      | |
 |  _ <  | |___  | . \  | |_| | |  _ <  |_____| | |___  | |___   | |
 |_| \_\ |_____| |_|\_\  \___/  |_| \_\          \____| |_____| |___|
rekor-cli: Rekor CLI

GitVersion:    v0.10.0
GitCommit:     83a4094253c9267d664dfddcf44ad5929a626bee
GitTreeState:  clean
BuildDate:     2022-07-29T10:11:30Z
GoVersion:     go1.18.4
Compiler:      gc
Platform:      darwin/arm64
```

**Comment by LeSuisse** [comment]
I'm confirming it works fine, thanks. I think we can close?
